### ISSUE TITLE
Reimport audio from source

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#778]
The sprite UI has a right click "reimport sprite from source", but there's nothing similar in the audio, currently you have to delete the audiocache folder and then press f7 to force an audio clip you updated to be reloaded.

**Comment by ivan-mogilko** [comment]
I thought it should do this automatically?

**Comment by ericoporto** [comment]
You mean, it will update whatever audio is different from cache automatically? When? I may have done something wrong then.

**Comment by morganwillcock** [comment]
I think it tracks file modification times, but from a quick search of the code it is only checked if you are using a separate file for your audio. It might be that unless you choose to the rebuild the game files it is still using what was already compiled into the exe.

**Comment by ericoporto** [comment]
I am not using separate audio files, you mean the VOX files right? I've built into the exe

**Comment by morganwillcock** [comment]
Yes. I think it doesn't check the timestamps if you aren't using VOX files.

**Comment by ericoporto** [comment]
I noticed this is a duplicate of #565

**Comment by ivan-mogilko** [comment]
@morganwillcock I forgot, was anything done in regards to this?

**Comment by morganwillcock** [comment]
I don't think anything was changed to try and fix it, no.

**Comment by morganwillcock** [comment]
I checked and it seems that all audioclips are updated in the exe file, they are just always included with no timestamp checks. Where there may be some confusion is that testing the game in the editor doesn't require the exe in the compiled folder to be in an updated state, in-fact the behaviour of the exe being deleted when testing the game from the editor protects you from accidentally distributing an exe with the wrong audio in it.

Where there is a problem; the same protection doesn't apply to the Linux and data file build targets. These are not removed when testing the game in the editor and will still contain the old audio data.

So you can go wrong with audio stored 'InGameEXE' by:
- updating the audio file in its source location
- testing the game in the editor (to hear your change)
- distributing the Linux build folder or .ags datafile without choosing to rebuild (the updated audio will not be present)

I would classify this as a bug so will leave this ticket open, although if you always rebuild the game before distributing you shouldn't see this issue.

**Comment by ivan-mogilko** [comment]
> I would classify this as a bug so will leave this ticket open, although if you always rebuild the game before distributing you shouldn't see this issue.

On contrary, I doubt this may be considered a bug. The state of whatever is in Compiled folder depends on running "Build Exe" / "Rebuild all files" command. Test run with F5 has nothing to do with the state of the Compiled folder in principle. Editor may be changed to not delete Windows exe inside compiled folder too (there was a request for that).

**Comment by morganwillcock** [comment]
I don't think other engines make it unclear what state the project is in. On one hand you have people wondering why there compiled versions are continually being deleted by the action of testing, on the other some compiled files are being left in-place and going out of sync with the source files.

Which is the correct approach is arguable, but what is not correct is the middle ground where no-one can tell what state the files are in and is forced to perform a manual rebuild to achieve a known state.

**Comment by ivan-mogilko** [comment]
> Which is the correct approach is arguable, but what is not correct is the middle ground where no-one can tell what state the files are

By "middle ground" do you mean that it deletes compiled files in one folder but not the other?

**Comment by ivan-mogilko** [comment]
I guess what I am arguing here is whether compiled folder may be treated as a part of the "project state".
I don't really know all the reasons why the game.exe is getting deleted when the test game is run. You mentioned it may be to force user to rebuild, but that seem to only "work" by assumption that user will test the game from the editor after any changes.

IMO there were other, technical, reasons to delete exe: because of the way engine searched for resources it used to check the project folder and "caches" latest, and that could cause it reading old versions of files from exe during test run.
This could've been coincidentally fixed by some of the recent changes to the engine (but I'd have to double check).

**Comment by morganwillcock** [comment]
I guess the easiest solution is define a directory (or archive location) as being for distribution only and then only the final publishing step creates resources here. Part of the confusion might be that a menu option called 'build exe' doesn't actually say why the exe is being built, and at other times an exe is built without choosing to 'build exe'. The directory name 'compiled' also doesn't describe the intent.

**Comment by ericoporto** [comment]
The request from the ticket diverges a lot from the scope of the final messages, and this is in 3.6.0 milestone for some reason. 

It's probably best to either review if this needs to be in 3.6.0 and figure a reduced scope or write a new ticket and close this.
